src/pg/pg-store.ts

```diff
@@ -478,6 +482,31 @@ export class PgStore extends BasePgStore {
     }
   }
 
+  async getInscriptionCountPerBlock(
+    filters?: DbInscriptionCountPerBlockFilters
+  ): Promise<DbInscriptionCountPerBlock[]> {
+    const fromCondition = filters?.from_block_height
+      ? this.sql`block_height >= ${filters?.from_block_height}`
+      : this.sql``;
+
+    const toCondition = filters?.to_block_height
+      ? this.sql`block_height <= ${filters?.to_block_height}`
+      : this.sql``;
+
+    const where =
+      filters?.from_block_height && filters?.to_block_height
+        ? this.sql`WHERE ${fromCondition} AND ${toCondition}`
+        : this.sql`WHERE ${fromCondition}${toCondition}`;
+
+    return await this.sql<DbInscriptionCountPerBlock[]>`
+      SELECT *
+      FROM inscriptions_per_block
+      ${filters?.from_block_height || filters?.to_block_height ? where : this.sql``}
+      ORDER BY block_height DESC
+      LIMIT 10000
```

Comment:
Do you still need this limit given that you're providing the block height range already? I think you should validate a max block range in the filters instead

Reply:
I kept the limit for when only one end of the range is given, but we could instead also generate the other end if none is given in the same step where we would validate.

Reply:
Sure, that sounds good. The idea is to prevent a query from selecting all available blocks which will shoot the DB CPU to the moon

Reply:
But that is essentially just selecting a different LIMIT value, right? If the params are set those values are used (and the limit ignored, assuming the params specify a smaller range); if a params (or both) is not set it would take the limit amount.
I assumed 10_000 was still a fairly doable number for database rows (since no additional calculation/join is being done.

Also thinking about the use-cases for the endpoint, ideally we could allow a full chart (since inscriptions were added) of statistics history.
The response could further be optimized by only returning the data/values in array format, rather than rows of named entries (but this is more concerning response size).

I'm happy to implement whichever is best 😉